### PR TITLE
Set cloud

### PR DESCRIPTION
**Pull request by amitlalani**

### CODE REVIEW

File: tests/scripts/test_config.py [hunk 1]
```diff
@@ -65,6 +68,16 @@ def test_set_regions_not_iterable(self, config_proxy, global_state):
...
+            'foo': 'bar',
+        }
+
+        assert config_proxy.get_cloud('foo') == 'bar'
+
+
+@patch('gonzo.config.ConfigProxy.CLOUDS', new_callable=PropertyMock)
 class TestAvailableClouds(object):
     def test_ok(self, clouds):
         clouds.return_value = {
```
Comment (ralka) [inline tests/scripts/test_config.py]: patching "CLOUDS" and passing "clouds"

**Comment by ralka** [comment]
:+1:

File: gonzo/config.py [hunk 2]
```diff
@@ -142,8 +143,8 @@ def get_namespaced_cloud_config_value(self,
         if override is not None:
             return override
 
-        # No override supplied, so check config.
-        namespaced_dict = self.CLOUD[config_key]
+        # No override supplied, so check configself.
```
Comment (noisyboiler) [inline gonzo/config.py]: typo? `configself`

File: gonzo/helpers/document_loader.py [hunk 3]
```diff
@@ -29,12 +29,12 @@ def build_params_dict(entity_name, config_params_key, additional_params=None):
     params = {
         'hostname': entity_name,
         'stackname': entity_name,
-        'domain': config.CLOUD['DNS_ZONE'],
-        'fqdn': "%s.%s" % (entity_name, config.CLOUD['DNS_ZONE']),
+        'domain': config.get_cloud()['DNS_ZONE'],
```
Comment (noisyboiler) [inline gonzo/helpers/document_loader.py]: could localise the result of this function call to save calling it 4 times?

**Pull request merged**